packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -122,9 +145,14 @@ export const ThemeProvider = React.forwardRef((props, forwardedRef) => {
       >
         <ToastProvider>
           {children}
-          <div ref={portalContainerRef}>
-            <Toaster />
-          </div>
+
+          {portalContainerProp ? (
+            ReactDOM.createPortal(<Toaster />, portalContainerProp)
+          ) : (
```

Comment:
When portaling a component, do we not also need to portal its Provider?

i.e. something like:

```tsx
{portalContainerProp ? (
  ReactDOM.createPortal(
    <ToastProvider>
      {children}
      <Toaster />
    </ToastProvider>, 
    portalContainerProp)
) : (
  <ToastProvider>
    {children}
    <div ref={setPortalContainerState}>
      <Toaster />
    </div>
  </ToastProvider>
)
}
```

In other words, can Toaster (or any other component for that matter) access its associated Provider if it has been portaled in such a way where the Provider is not an ancestor of that component?

Reply:
the provider gets automatically connected. see the second paragraph here: https://react.dev/reference/react-dom/createPortal#reference